File: docs/guides/examples/rs/examples/data_querying_timestamp.rs
use std::time::SystemTime;

use bytes::Bytes;
use reduct_rs::{ReductClient, ReductError};
use tokio;

#[tokio::main]
async fn main() -> Result<(), ReductError> {
    // Create a client instance, then get or create a bucket
    let client = ReductClient::builder()
        .url("http://127.0.0.1:8383")
        .api_token("my-token")
        .build();
    let bucket = client.create_bucket("test").exist_ok(true).send().await?;

    // Send a record to the "entry-1" entry with the current timestamp
    let timestamp = SystemTime::now();
    bucket
        .write_record("entry-1")
        .timestamp(timestamp)
        .data(Bytes::from("Some binary data"))
        .send()
        .await?;

    // Query records in the time range
    let record = bucket
        .read_record("entry-1")
        .timestamp(timestamp).send().await?;

    println!("Timestamp: {:?}", record.timestamp());
    println!("Content Length: {}", record.content_length());
    println!("Content Type: {}", record.content_type());
    println!("Labels: {:?}", record.labels());

    // Read the record data
    let data = record.bytes().await?;
    assert_eq!(data, Bytes::from("Some binary data"));


    Ok(())
}

File: versioned_docs/version-1.10.x/examples/py/src/data_querying_timestamp.py
import time
import asyncio
from reduct import Client, Bucket


async def main():
    # Create a client instance, then get or create a bucket
    client = Client("http://127.0.0.1:8383", api_token="my-token")
    bucket: Bucket = await client.create_bucket("my-bucket", exist_ok=True)

    ts = time.time()
    await bucket.write(
        "py-example",
        b"Some binary data",
        ts,
    )

    # Query records in the "py-example" entry of the bucket
    async with bucket.read("py-example", ts) as record:
        # Print  meta information
        print(f"Timestamp: {record.timestamp}")
        print(f"Content Length: {record.size}")
        print(f"Content Type: {record.content_type}")
        print(f"Labels: {record.labels}")

        # Read the record content
        content = await record.read_all()
        assert content == b"Some binary data"


loop = asyncio.get_event_loop()
loop.run_until_complete(main())
